clouddrift/analysis.py

```diff
@@ -40,15 +44,18 @@ def apply_ragged(
     ----------
     func : callable
         Function to apply to each row of each ragged array in ``arrays``.
-    arrays : list[np.ndarray] or np.ndarray
+    arrays : list[np.ndarray] or np.ndarray or xr.DataArray
         An array or a list of arrays to apply ``func`` to.
-    rowsize : list
+    rowsize : list[int] or np.ndarray[int] or xr.DataArray[int]
         List of integers specifying the number of data points in each row.
     *args : tuple
         Additional arguments to pass to ``func``.
     rows : int or Iterable[int], optional
         The row(s) of the ragged array to apply ``func`` to. If ``rows`` is
```

Comment:
The fact that the default `None` means apply to all rows is very counter-intuitive to me.

Reply:
I can see it being counter-intuitive because "None" for "rows" implies "apply to no rows". However, this is a Python quirk--`None` is a special word that in this case means that the `rows` argument. Though at the risk of being counter-intuitive at first, it is _the_ Pythonic solution.